File: 6_kyu/Does_my_number_look_big_in_this.java
public class NumberUtils {

    public static boolean isNarcissistic(int number) {
        
        String numero = String.valueOf(number);
        int tamaño = numero.length();
        int suma = 0;

        for (int i = 0; i < tamaño; i++) {
            suma += Math.pow(Character.getNumericValue(numero.charAt(i)), tamaño);
        }

        return suma == number ? true : false;
    }

}

File: 6_kyu/Does_my_number_look_big_in_this.php
<?php

function narcissistic(int $value): bool {
    $numero = strval($value);
    $longitud = strlen($numero);
    $suma = 0;

    for ($i=0; $i < $longitud ; $i++) { 
        $suma += pow($numero[$i], $longitud);
    }

    return $value === $suma;
}
